File: backend/models/photo.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from backend.database.base import Base


class Photo(Base):
    __tablename__ = 'photos'

    id = Column(Integer, primary_key=True, index=True)
    claim_id = Column(Integer, ForeignKey('claims.id'), nullable=False)
    image_url = Column(String, default='')
    timestamp = Column(String, default='')
    gps_location = Column(String, default='')
    photo_type = Column(String, nullable=False)

    claim = relationship('Claim', back_populates='photos')

File: lib/models/photo.dart
class InspectionPhoto {
  final int id;
  final int claimId;
  final String imageUrl;
  final String timestamp;
  final String gpsLocation;
  final String photoType;

  InspectionPhoto({
    required this.id,
    required this.claimId,
    required this.imageUrl,
    required this.timestamp,
    required this.gpsLocation,
    required this.photoType,
  });

  factory InspectionPhoto.fromJson(Map<String, dynamic> json) =>
      InspectionPhoto(
        id: json['id'] as int? ?? 0,
        claimId: json['claim_id'] as int? ?? 0,
        imageUrl: json['image_url'] as String? ?? '',
        timestamp: json['timestamp'] as String? ?? '',
        gpsLocation: json['gps_location'] as String? ?? '',
        photoType: json['photo_type'] as String? ?? '',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'claim_id': claimId,
        'image_url': imageUrl,
        'timestamp': timestamp,
        'gps_location': gpsLocation,
        'photo_type': photoType,
      };
}
